docs/topics/migrations.txt

```diff
@@ -622,26 +622,136 @@ to be able to write your own, see the :doc:`migration operations reference
 </ref/migration-operations>` and the "how-to" on :doc:`writing migrations
 </howto/writing-migrations>`.
 
-.. _migration-squashing:
+.. _migration-trimming:
 
-Squashing migrations
-====================
+Trimming migrations
+===================
 
 You are encouraged to make migrations freely and not worry about how many you
 have; the migration code is optimized to deal with hundreds at a time without
 much slowdown. However, eventually you will want to move back from having
-several hundred migrations to just a few, and that's where squashing comes in.
+several hundred migrations to just a few. That's where trimming migrations
+comes in.
 
-Squashing is the act of reducing an existing set of many migrations down to
+Trimming is the act of reducing an existing set of many migrations down to
 one (or sometimes a few) migrations which still represent the same changes.
 
-Django does this by taking all of your existing migrations, extracting their
-``Operation``\s and putting them all in sequence, and then running an optimizer
-over them to try and reduce the length of the list - for example, it knows
-that :class:`~django.db.migrations.operations.CreateModel` and
-:class:`~django.db.migrations.operations.DeleteModel` cancel each other out,
-and it knows that :class:`~django.db.migrations.operations.AddField` can be
-rolled into :class:`~django.db.migrations.operations.CreateModel`.
+There are two common approaches to trimming migrations. The first is to use the
+``squashmigrations`` command to create a merged migration file, however this
+approach can be impractical on complex projects. The second approach is to
+coordinate with your team to ensure that all installations of your app are up
+to date, then to have a scheduled time when migrations are removed and
+recreated from scratch.
+
+Which approach is best for your project will depend on the complexity of it
+and the flexibility of your team.
+
+
+.. _manual-migration-trimming:
+
+Trimming migrations manually
+----------------------------
+
+The easiest and most reliable way to trim migrations is to do so manually,
+instead of using the ``squashmigrations`` command. When doing this process, any
+apps that have foreign key interdependencies will need to be trimmed at the same
+time so that old migrations do not depend on deleted migrations. For example,
+if your ``pizza`` app has a ``ForeignKey`` to an ``Ingredients`` model in your
+``food`` app, you will need to manually trim both the ``food`` and the
+``pizza`` apps simultaneously. It is often easiest to trim all apps in your
+project at the same time.
```

Comment:
I think this needs an **introductory paragraph** saying what you're going to do, then this warning about the need to be up to date, then the steps.